hnn_core/batch_simulate.py

```diff
@@ -307,13 +356,135 @@ def _save(self, results, start_idx, end_idx):
         if not os.path.exists(self.save_folder):
             os.makedirs(self.save_folder)
 
-        sim_data = np.stack([dpl['dpl'][0].data['agg'] for dpl in results])
+        save_data = {
+            'param_values': [result['param_values'] for result in results]
+        }
+
+        attributes_to_save = ['dpl', 'spiking', 'lfp',
+                              'voltages', 'currents', 'calcium']
+        for attr in attributes_to_save:
+            if getattr(self, f'save_{attr}') and attr in results[0]:
+                save_data[attr] = [result[attr] for result in results]
 
         file_name = os.path.join(self.save_folder,
-                                 f'sim_run_{start_idx}-{end_idx}.npy')
+                                 f'sim_run_{start_idx}-{end_idx}.npz')
         if os.path.exists(file_name) and not self.overwrite:
             raise FileExistsError(
                 f"File {file_name} already exists and "
                 "overwrite is set to False.")
 
-        np.save(file_name, sim_data)
+        np.savez(file_name, **save_data)
+
+    def _save_parameters(self, param_combinations, start_idx, end_idx):
```

Comment:
Decide if you want to have a single file both data and parameters, or split between a data file and a dedicated parameters file